### PR TITLE
checks if link points to valid url

### PR DESCRIPTION
**Pull request by pkthewizard**
#64 

-added function to check if url string points to an actual URL and added to right click

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -18,6 +18,18 @@ chrome.contextMenus.removeAll(function() {
 chrome.contextMenus.onClicked.addListener(function(info){
     let link:string = info.linkUrl as string;
 
-    //Updates the URL Array stored in memory with the new link
-    URLStorage.putURL(link);
-});
\ No newline at end of file
+    //Updates the URL Array stored in memory with the new link if link points to valid file
+    if(validateURL(link)) 
+    {
```
Comment (kpattaswamy) [inline src/background.ts:23]: move bracket to prev line

File: src/background.ts [hunk 2]
```diff
@@ -18,6 +18,18 @@ chrome.contextMenus.removeAll(function() {
...
+        URLStorage.putURL(link);
+    }
+});
+
+function validateURL(url:string) {
+    try {
+        new URL(url);
+        return true;
```
Comment (kpattaswamy) [inline src/background.ts:31]: return true at the end of the function

File: src/background.ts [hunk 3]
```diff
@@ -18,6 +18,18 @@ chrome.contextMenus.removeAll(function() {
...
-});
\ No newline at end of file
+    //Updates the URL Array stored in memory with the new link if link points to valid file
+    if(validateURL(link)) 
+    {
+        URLStorage.putURL(link);
+    }
+});
+
+function validateURL(url:string) {
```
Comment (kpattaswamy) [inline src/background.ts:28]: can you make this function exportable

**Review by kpattaswamy — APPROVED** [review]
Im assuming this is tested and works

Do you think you it would be possible to add functionality to check what kind of file exists at the URL?

**Comment by pkthewizard** [comment]
Should this only return true if it has a pdf extension? It works for any URL right now, but I can make it work for PDFs.

**Review by kpattaswamy — APPROVED** [review]
nice

**Pull request merged**